test/integration/authorization_test.go

```diff
@@ -11,17 +11,105 @@ import (
 	policy "github.com/openshift/origin/pkg/cmd/admin/policy"
 	"github.com/openshift/origin/pkg/cmd/server/bootstrappolicy"
+	imageapi "github.com/openshift/origin/pkg/image/api"
+	oauthapi "github.com/openshift/origin/pkg/oauth/api"
 	testutil "github.com/openshift/origin/test/util"
 	testserver "github.com/openshift/origin/test/util/server"
 )
 
+func TestClusterReaderCoverage(t *testing.T) {
+	testutil.RequireEtcd(t)
+	_, clusterAdminKubeConfig, err := testserver.StartTestMasterAPI()
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+
+	clusterAdminClient, err := testutil.GetClusterAdminClient(clusterAdminKubeConfig)
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+
+	discoveryClient := client.NewDiscoveryClient(clusterAdminClient.RESTClient)
+
+	// (map[string]*unversioned.APIResourceList, error)
+	allResourceList, err := discoveryClient.ServerResources()
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+
+	allResources := map[unversioned.GroupResource]bool{}
+	for _, resources := range allResourceList {
+		version, err := unversioned.ParseGroupVersion(resources.GroupVersion)
+		if err != nil {
+			t.Fatalf("unexpected error: %v", err)
+		}
+
+		for _, resource := range resources.APIResources {
+			allResources[version.WithResource(resource.Name).GroupResource()] = true
+		}
+	}
+
+	escalatingResources := map[unversioned.GroupResource]bool{
+		oauthapi.Resource("oauthauthorizetokens"): true,
+		oauthapi.Resource("oauthaccesstokens"):    true,
+		oauthapi.Resource("oauthclients"):         true,
+		imageapi.Resource("imagestreams/secrets"): true,
+		kapi.Resource("secrets"):                  true,
+		kapi.Resource("pods/exec"):                true,
+		kapi.Resource("pods/proxy"):               true,
+		kapi.Resource("pods/portforward"):         true,
+		kapi.Resource("nodes/proxy"):              true,
+		kapi.Resource("services/proxy"):           true,
+	}
+
+	readerRole, err := clusterAdminClient.ClusterRoles().Get(bootstrappolicy.ClusterReaderRoleName)
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+	for _, rule := range readerRole.Rules {
+		for _, group := range rule.APIGroups {
+			for resource := range rule.Resources {
+				gr := unversioned.GroupResource{Group: group, Resource: resource}
+				if escalatingResources[gr] {
+					t.Errorf("cluster-reader role has escalating resource %v.  Check pkg/cmd/server/bootstrappolicy/policy.go.", gr)
+				}
+				delete(allResources, gr)
+			}
+		}
+	}
+
+	// remove escalating resources that cluster-reader should not have access to
+	for resource := range escalatingResources {
+		delete(allResources, resource)
+	}
+
+	// remove resources without read APIs
+	nonreadingResources := []unversioned.GroupResource{
+		buildapi.Resource("buildconfigs/instantiatebinary"), buildapi.Resource("buildconfigs/instantiate"), buildapi.Resource("builds/clone"),
+		imageapi.Resource("imagestreamimports"),
+		extensionsapi.Resource("deployments/rollback"),
```

Comment:
We will have this with https://github.com/openshift/origin/pull/9135. You need `deploymentconfigrollbacks` and I also didn't see `generatedeploymentconfigs`.